cores/esp8266/core_esp8266_i2s.c

```diff
@@ -92,10 +95,16 @@ void ICACHE_FLASH_ATTR i2s_slc_isr(void) {
   }
 }
 
-void ICACHE_FLASH_ATTR i2s_slc_begin(){
+void ICACHE_FLASH_ATTR i2s_slc_begin(uint16_t buffers, uint16_t buffer_samples){
+  SLC_BUF_CNT = buffers;
+  SLC_BUF_LEN = buffer_samples;
+
   i2s_slc_queue_len = 0;
   int x, y;
   
+  i2s_slc_queue = (uint32_t*)malloc((SLC_BUF_CNT-1) * sizeof(uint32_t));
```

Comment:
Is it correct that this malloc is 1 item smaller than the others? I realize this is how it is currently implemented, but I have to wonder.

Reply:
Before the malloc is done, the pointer should be checked against null, and free()d if not.

Reply:
SIze is -1 by design (see below, the original code only alloc'd buf_cnt-1).  Good catch on 2nd comment.  It's safe to free(NULL), so I'll always free before allocing.

Reply:
You probably don't want to free and malloc again if the call is done with the same size arg.